app/src/main/java/com/github/se/bootcamp/ui/navigation/NavigationActions.kt

```diff
@@ -6,17 +6,35 @@ import androidx.navigation.NavHostController
 open class NavigationActions(
     private val navController: NavHostController,
 ) {
-    fun navigateTo(destination: TopLevelDestination) {
+    open fun navigateTo(destination: TopLevelDestination) {
+
         navController.navigate(destination.route) {
+            // Pop up to the start destination of the graph to
+            // avoid building up a large stack of destinations
             popUpTo(navController.graph.findStartDestination().id) {
                 saveState = true
+                inclusive = true
             }
+
+            // Avoid multiple copies of the same destination when reselecting same item
             launchSingleTop = true
-            restoreState = true
+
+            // Restore state when reselecting a previously selected item
+            //if (destination.route != Route.AUTH) {
+              //  restoreState = true
+            //}
```

Comment:
Ideally, we should avoid merging commented code into the main branch. Commented code tends to stay longer than expected and can accumulate technical debt. If this code is necessary for future work, it might be better to track it in a task or discussion, or remove it until it's ready for use.